### ISSUE TITLE
Prisma is not compatible with Next.js webpack5 opt-in

### ISSUE THREAD

**Issue by meeq** [prisma/prisma#6328]
## Bug description

Next.js recently announced the move to Webpack 5, which is supported by opting-in with `next.config.js`: https://nextjs.org/docs/messages/webpack5

Prisma fails to initialize when used with Next.js when `future.webpack5` is `true`:

```
 API error: PrismaClientInitializationError2 [PrismaClientInitializationError]: 
 Invalid `prisma.account.findUnique()` invocation:
 
 
   Query engine exited with code 101
 
 thread 'main' panicked at 'Could not open datamodel file "/app/.next/server/pages/api/Account/schema.prisma"', query-engine/query-engine/src/opt.rs:187:53
 stack backtrace:
    0: rust_begin_unwind
              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:493:5
    1: std::panicking::begin_panic_fmt
              at /rustc/2fd73fabe469357a12c2c974c140f67e7cdd76d0/library/std/src/panicking.rs:435:5
    2: query_engine::opt::load_datamodel_file::{{closure}}
    3: structopt::StructOpt::from_args
    4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
    5: std::thread::local::LocalKey<T>::with
    6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
    7: async_io::driver::block_on
    8: async_global_executor::reactor::block_on
    9: std::thread::local::LocalKey<T>::with
   10: async_std::task::builder::Builder::blocking
   11: query_engine::main
 note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
     at cb (webpack-internal:///./prisma/client/runtime/index.js:410:539)
     at runMicrotasks (<anonymous>)
     at processTicksAndRejections (internal/process/task_queues.js:93:5)
     at async GET (webpack-internal:///./src/pages/api/Account/[id].ts:44:19)
     at async eval (webpack-internal:///./src/server/api.ts:50:16)
     at async apiResolver (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/api-utils.js:8:1)
     at async DevServer.handleApiRequest (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/next-server.js:67:462)
     at async Object.fn (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/next-server.js:59:492)
     at async Router.execute (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/router.js:25:67)
     at async DevServer.run (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/next-server.js:69:1042)
     at async DevServer.handleRequest (/app/.yarn/$$virtual/next-virtual-32fc5bd849/0/cache/next-npm-10.1.2-24c2718295-97ce484973.zip/node_modules/next/dist/next-server/server/next-server.js:34:504) {
   clientVersion: '2.20.0',
   errorCode: undefined
 }
```

Prisma appears to be attempting to read the schema file from the wrong place.

## How to reproduce

1. Create Next.js project
2. Opt-in to webpack5 (see https://nextjs.org/docs/messages/webpack5)
3. Initialize Prisma 2.20.0 in project
4. Query database using Prisma Client in Next.js page

## Expected behavior

Prisma should not catastrophically fail to initialize

## Environment & setup

- OS: Debian
- Database: Postgres 13.2
- Node.js version: 14.16.0
- Prisma version: 2.20.0

**Comment by millsp** [comment]
We tried to reproduce this error, could you provide a repro please?

**Comment by litewarp** [comment]
So I'm having this issue too on a nexjs app, but based on @millsp's comment, I tried to recreate the error from the rest-nextjs-api-routes example. But I was unable to reproduce the error there as well.

So I'm still getting this error when I try to use webpack5 in my app:
```
thread 'main' panicked at 'Could not open datamodel file "/Users/litewarp/Projects/conflicto/.next/server/pages/schema.prisma"', query-engine/query-engine/src/opt.rs:187:53
```

I have some differences between a vanilla setup, such as some plugins and a custom path for my schema file. The problem probably lies somewhere in there.

**Comment by meeq** [comment]
I found a work-around for this. Webpack must treat the prisma client as an external: https://webpack.js.org/configuration/externals/#function

My `next.config.js` looks something like:
```js
module.exports = {
  future: {
    webpack5: true,
  },
  webpack: (config, { isServer }) => {
    if (isServer) {
      config.externals.unshift(({ context, request }, callback) => {
        // Work-around for Prisma "Could not open datamodel file"  error
        if (request === 'prisma/client') {
          return callback(null, `commonjs ${prismaClientPath}`);
        }
        if (request === './runtime' && context === prismaClientPath) {
          return callback(null, `commonjs ${prismaClientPath}/runtime`);
        }
        callback();
      });
    }
    return config;
  },
};
```

**Comment by millsp** [comment]
@meeq Do you still get this error in `2.21.1`? If yes, could you provide a repro? :pray:

**Comment by millsp** [comment]
I got a reproduction locally, working on a fix.

**Comment by cyrus-za** [comment]
I am running into this error on next 10.1.3 with webpack 4.

I am using a [custom output directory](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#using-a-custom-output-path) and somehow it thinks that the root that I am running it from is where the output is relative to.

I am using a [nx monorepo](https://github.com/nrwl/nx) and my prisma schema and client is in a lib. I've set the output in my schema to a custom dir
```prisma
generator client {
  provider = "prisma-client-js"
  output   = "./generated/client"
}
```

It now tries to go from `root/src/generated/client` where it should be from `root/libs/db/prisma/generated/client`

I get this error
```sh
thread 'main' panicked at 'Could not open datamodel file "root/src/generated/client/schema.prisma"', query-engine/query-engine/src/opt.rs:187:53
```
NOTE: I replaced my monorepo working directory with `root` instead of `/Users/...` to simplify explanation
